postgresql: some improvements to the derivation

wolfgangwalther opened the pull request:
## Description of changes

This fell out of my work on "building with meson / in pkgsStatic", which I have recently been progressing well with. Still nothing to show, yet, though.

@Ma27 let me know what you think!

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

wolfgangwalther committed:
pkgsMusl.postgresql: disable "locale -a" properly

/bin/locale doesn't exist on musl and was already effectively disabled
in #228349. However this still leaves the following warning for initdb:

  performing post-bootstrap initialization ... sh: locale: not found

By applying the alpine patch to disable locale -a entirely, this warning
will disappear. This will also make one more regression test pass when
testing "check-world" instead of "check", only.

wolfgangwalther committed:
postgresql: run full test-suite during checkPhase

doCheck was enabled in #61395 which does not mention an explicit decision
for "check" instead of "check-world". The latter runs more tests, for example
for contrib modules etc. - but results in higher build times..

Running "nix-build -A postgresql":
- with doCheck = false: 46s
- with checkTarget = "check": 60s
- with checkTarget = "check-world": 75s

wolfgangwalther committed:
postgresql: simplify systemdSupport default value

systemd is not available on static platforms, so the isStatic check is
redundant.

wolfgangwalther committed:
postgresql: use newer libuuid instead of ossp-uuid

Since PostgreSQL 9.4 it's possible to build with different uuid
libraries. Eversince, the postgres docs [1] have this to say:

  [..] the OSSP UUID library [..] is not well maintained, and is
  becoming increasingly difficult to port to newer platforms. [..]
  On Linux, OS X, and some other platforms, suitable functions are
  provided in the libuuid library, which originally came from the
  e2fsprogs project (though on modern Linux it is considered part of
  util-linux-ng).

Using e2fs for Darwin was introduced 9 years ago in #8574, which library
to use for Linux was never considered again. Thus, start building with
newer libs now.

As a side effect, we require one less patch if we decide to build with
meson later.

[1]: https://www.postgresql.org/docs/current/uuid-ossp.html

doronbehar reviewed (approved):
Changes look very good and are well documented. Should be good to go once CI is green.

Ma27 commented:
Sure, will do so tomorrow.

wolfgangwalther committed:
postgresql: remove deprecated enableSystemd override

This was deprecated in e6bfabf366b7dfdf67e24877dc5a24c5c1f8e62e, where
we agreed on removing this after one release in [1].

Time has come!

[1]: https://github.com/NixOS/nixpkgs/pull/292993#discussion_r1518883146

doronbehar commented:
Oh but there is a merge conflict.

doronbehar commented:
Could you please verify that on the rebased branch everything works at least on your platform? ofborg's logs are not green and it is not clear to me why.

doronbehar commented:
> Imho, ofborg had been failing before, too. I did verify the changes on master only, so far. I don't expect any change after the kind of conflicts I fixed.
> 
> I will kick off another build on staging though. Might take a while ;)

Oh I see, I haven't noticed we are targeting staging, so there is no need for that. In fact, I think this PR is mature enough to be further cooked in staging :).

doronbehar commented:
Aish I would have merged this right now if you hadn't forced pushed. Let's merge this after evaluation is green.

Ma27 commented on pkgs/servers/sql/postgresql/generic.nix:
```diff
@@ -102,16 +98,17 @@ let
       ++ lib.optionals gssSupport [ libkrb5 ]
       ++ lib.optionals stdenv'.hostPlatform.isLinux [ linux-pam ]
-      ++ lib.optionals (!stdenv'.hostPlatform.isDarwin) [ libossp_uuid ];
+      ++ lib.optionals (!stdenv'.hostPlatform.isDarwin) [ e2fsprogs ];
```
Can you explain why both `libossp_uuid` & `e2fsprogs` are needed? From what I've read in the commit message, only one of them is required?

Ma27 replied:
> libossp_uuid is removed, but you mean libuuid (further up) and e2fsprogs, I guess.

Correct, good morning ma27 🙃

Ma27 reviewed (approved):
Thank _you_!
